app/javascript/__tests__/Articles/ArticlesDetails.test.tsx

```diff
@@ -1,29 +1,111 @@
-  container = document.createElement("div");
-  document.body.appendChild(container);
-});
+describe("ArticleDetails", () => {
+  describe("Snapshots", () => {
+    test("should render the Article Details when logged out", () => {
+      const { container } = render(
+        <ArticleDetails
+          title={expectedTitle}
+          content={expectedContent}
+          firstName={expectedFirstName}
+          lastName={expectedLastName}
+          link={expectedLink}
+          linkText={expectedLinkText}
+          createdAt={expectedCreatedAt}
+        />
+      );
+      expect(container).toMatchSnapshot();
+    });
 
-afterEach(() => {
-  unmountComponentAtNode(container);
-  container.remove();
-  container = null;
-});
+    test("should render the Article Details when logged in", () => {
+      const { container } = render(
+        <ArticleDetails
+          title={expectedTitle}
+          content={expectedContent}
+          firstName={expectedFirstName}
+          lastName={expectedLastName}
+          link={expectedLink}
+          linkText={expectedLinkText}
+          createdAt={expectedCreatedAt}
+          editLink={expectedEditLink}
+          currentUser={expectedCurrentUser}
+        />
+      );
+      expect(container).toMatchSnapshot();
+    });
+  });
+
+  describe("Props", () => {
+    test("should render the text passed in", () => {
+      const { container } = render(
+        <ArticleDetails
+          title={expectedTitle}
+          content={expectedContent}
+          firstName={expectedFirstName}
+          lastName={expectedLastName}
+          link={expectedLink}
+          linkText={expectedLinkText}
+          createdAt={expectedCreatedAt}
+        />
+      );
+      expect(container).toHaveTextContent(expectedTitle);
+      expect(container).toHaveTextContent(expectedContent);
+      expect(container).toHaveTextContent(expectedFirstName);
+      expect(container).toHaveTextContent(expectedLastName);
+      expect(container).toHaveTextContent(expectedLinkText);
+      expect(container).toHaveTextContent(expectedCreatedAt);
+    });
+
+    test("should render the hrefs passed in", () => {
+      const { getByText } = render(
+        <ArticleDetails
+          title={expectedTitle}
+          content={expectedContent}
+          firstName={expectedFirstName}
+          lastName={expectedLastName}
+          link={expectedLink}
+          linkText={expectedLinkText}
+          createdAt={expectedCreatedAt}
+        />
+      );
+      expect(getByText(expectedLinkText).href).toBe(
+        `http://localhost${expectedLink}`
+      );
+    });
 
-it('renders correctly', ()=>{
-  const tree = renderer.create(<ArticleDetails 
-    title='New Post' 
-    content='Ipsum Lorem'
-    firstName='Megan' 
-    link='google.com' 
-    linkText='Click Here' 
-    created_at='27 March 2020' />).toJSON();
-  expect(tree).toMatchSnapshot();
+    test("should render the edit option text if a current user is passed", () => {
```

Comment:
as a suggestion could have another test here: does not render if the edit button if the current user is not passed in.

Reply:
Good idea! That's been added in now :)